<a href="https://colab.research.google.com/github/cherishkim42/covid/blob/master/ItalyEDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
from google.colab import drive
drive.mount('/content/gdrive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/gdrive


1. Which regions were most afflicted by COVID-19?
1. On which days were infection rates the most severe? Least severe?
1. What is the fatality rate?



In [4]:
from scipy import stats
import math
import csv
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
sns.set(rc={'figure.figsize':(15,11)}) #standardize sizing for seaborn plots

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


# Italy

### Questions to address:

1. Which regions were most afflicted by COVID-19?
1. On which days was the number of new positive cases highest? Lowest?
1. What is the fatality rate?

### Which regions were most afflicted by COVID-19?

In [17]:
region = pd.read_csv("/content/gdrive/My Drive/Colab Notebooks/data/italy/covid19_italy_region.csv")
region.head()

SNo                 Date Country  ...  Deaths TotalPositiveCases  TestsPerformed
0    0  2020-02-24T18:00:00     ITA  ...       0                  0             NaN
1    1  2020-02-24T18:00:00     ITA  ...       0                  0             NaN
2    2  2020-02-24T18:00:00     ITA  ...       0                  0             NaN
3    3  2020-02-24T18:00:00     ITA  ...       0                  0             NaN
4    4  2020-02-24T18:00:00     ITA  ...       0                 18             NaN

[5 rows x 17 columns]

In [19]:
region.dtypes

SNo                            int64
Date                          object
Country                       object
RegionCode                     int64
RegionName                    object
Latitude                     float64
Longitude                    float64
HospitalizedPatients           int64
IntensiveCarePatients          int64
TotalHospitalizedPatients      int64
HomeConfinement                int64
CurrentPositiveCases           int64
NewPositiveCases               int64
Recovered                      int64
Deaths                         int64
TotalPositiveCases             int64
TestsPerformed               float64
dtype: object

In [21]:
region['RegionName'].unique()

array(['Abruzzo', 'Basilicata', 'Calabria', 'Campania', 'Emilia-Romagna',
       'Friuli Venezia Giulia', 'Lazio', 'Liguria', 'Lombardia', 'Marche',
       'Molise', 'P.A. Bolzano', 'P.A. Trento', 'Piemonte', 'Puglia',
       'Sardegna', 'Sicilia', 'Toscana', 'Umbria', "Valle d'Aosta",
       'Veneto'], dtype=object)

In [27]:
region['NewPositiveCases'].sum()

244666

In [24]:
abruzzo = region[(region['RegionName'] == 'Abruzzo')]
basilicata = region[(region['RegionName'] == 'Basilicata')]
calabria = region[(region['RegionName'] == 'Calabria')]
campania = region[(region['RegionName'] == 'Campania')]
emilia = region[(region['RegionName'] == 'Emilia-Romagna')]
friuli = region[(region['RegionName'] == 'Friuli Venezia Giulia')]
lazio = region[(region['RegionName'] == 'Lazio')]
liguria = region[(region['RegionName'] == 'Liguria')]
lombardia = region[(region['RegionName'] == 'Lombardia')]
marche = region[(region['RegionName'] == 'Marche')]
molise = region[(region['RegionName'] == 'Molise')]
pab = region[(region['RegionName'] == 'P.A. Bolzano')]
pat = region[(region['RegionName'] == 'P.A. Trento')]
piemonte = region[(region['RegionName'] == 'Piemonte')]
puglia = region[(region['RegionName'] == 'Puglia')]
sardegna = region[(region['RegionName'] == 'Sardegna')]
sicilia = region[(region['RegionName'] == 'Sicilia')]
toscana = region[(region['RegionName'] == 'Toscana')]
umbria = region[(region['RegionName'] == 'Umbria')]
valle = region[(region['RegionName'] == "Valle d'Aosta")]
veneto = region[(region['RegionName'] == 'Veneto')]

In [28]:
print("TOTAL POSITIVE CASES:")
print("Abruzzo: ", abruzzo['NewPositiveCases'].sum())
print("Basilicata: ", basilicata['NewPositiveCases'].sum())
print("Calabria: ", calabria['NewPositiveCases'].sum())
print("Campania: ", campania['NewPositiveCases'].sum())
print("Emilia-Romagna: ", emilia['NewPositiveCases'].sum())
print("Friuli Venezia Giulia: ", friuli['NewPositiveCases'].sum())
print("Lazio: ", lazio['NewPositiveCases'].sum())
print("Liguria: ", liguria['NewPositiveCases'].sum())
print("Lombardia: ", lombardia['NewPositiveCases'].sum())
print("Marche: ", marche['NewPositiveCases'].sum())
print("Molise: ", molise['NewPositiveCases'].sum())
print("P.A. Bolzano: ", pab['NewPositiveCases'].sum())
print("P.A. Trento: ", pat['NewPositiveCases'].sum())
print("Piemonte: ", piemonte['NewPositiveCases'].sum())
print("Puglia: ", puglia['NewPositiveCases'].sum())
print("Sardegna: ", sardegna['NewPositiveCases'].sum())
print("Sicilia: ", sicilia['NewPositiveCases'].sum())
print("Toscana: ", toscana['NewPositiveCases'].sum())
print("Umbria: ", umbria['NewPositiveCases'].sum())
print("Valle d'Aosta: ", valle['NewPositiveCases'].sum())
print("Veneto: ", veneto['NewPositiveCases'].sum())

TOTAL POSITIVE CASES:
Abruzzo:  3347
Basilicata:  409
Calabria:  1240
Campania:  4833
Emilia-Romagna:  29241
Friuli Venezia Giulia:  3356
Lazio:  8449
Liguria:  10105
Lombardia:  95542
Marche:  6813
Molise:  446
P.A. Bolzano:  2683
P.A. Trento:  4502
Piemonte:  31537
Puglia:  4559
Sardegna:  1388
Sicilia:  3541
Toscana:  10375
Umbria:  1456
Valle d'Aosta:  1196
Veneto:  19648


The hardest-hit regions of Italy so far are the following:


*   **Lombardia** with **95,542** positive cases;
*   **Piemonte** with **31,537** positive cases;
*   **Emilia-Romagna** with **29,241** positive cases;
*   **Veneto** with **19,648** positive cases;
*   **Toscana** with **10,375** positive cases; and
*   **Liguria** with **10,105** positive cases.

**Basilicata** is the least-hard hit region at **409** total positive cases.

### On which days were the infection rates most severe? Least severe?

In [31]:
(lombardia.sort_values('NewPositiveCases')).head(3)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
50      50  2020-02-26T18:00:00  ...                258             NaN
2969  2969  2020-07-14T17:00:00  ...              95173        693556.0
3074  3074  2020-07-19T17:00:00  ...              95492        721640.0

[3 rows x 17 columns]

In [32]:
(lombardia.sort_values('NewPositiveCases')).tail(3)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
680  680  2020-03-27T17:00:00  ...              37298             NaN
659  659  2020-03-26T17:00:00  ...              34889             NaN
554  554  2020-03-21T17:00:00  ...              25515             NaN

[3 rows x 17 columns]

In [36]:
(basilicata.sort_values('NewPositiveCases')).head(5)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
1534  1534  2020-05-07T17:00:00  ...                383         15780.0
1555  1555  2020-05-08T17:00:00  ...                382         16285.0
2311  2311  2020-06-13T17:00:00  ...                401         33891.0
2290  2290  2020-06-12T17:00:00  ...                401         33587.0
2248  2248  2020-06-10T17:00:00  ...                400         32761.0

[5 rows x 17 columns]

In [35]:
(basilicata.sort_values('NewPositiveCases')).tail(3)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
631  631  2020-03-25T17:00:00  ...                113             NaN
652  652  2020-03-26T17:00:00  ...                134             NaN
694  694  2020-03-28T17:00:00  ...                182             NaN

[3 rows x 17 columns]

For the hardest-hit region of **Lombardia**:
* The **most severe** days were:
   * **March 21st, 2020** with **3,251** new cases;
   * **March 26th, 2020** with **2,543** new cases;
   * **March 27th, 2020** with **2,409** new cases.
* The **least severe** days were:
   * **February 26th, 2020** with **18** new cases;
   * **July 14th, 2020** with **30** new cases;
   * **July 19th, 2020** with **33** new cases.

For the least-hard hit region of **Basilicata**:
* The **most severe** days were:
  * **March 28th, 2020** with **31** new cases;
  * **March 26th, 2020** with **21** new cases;
  * **March 25th, 2020** with **21** new cases.
* The **least severe days** were:
  * **June 10th, 12th, & 13th** with **0** new cases.
  * _There are two rows on the dataframe with negative values in the column for "NewPositiveCases"; this must be in error_

### What is the fatality rate?

In [40]:
(lombardia.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3095  3095  2020-07-20T17:00:00  ...              95548        724365.0

[1 rows x 17 columns]

In [41]:
(piemonte.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3100  3100  2020-07-20T17:00:00  ...              31537        288837.0

[1 rows x 17 columns]

In [42]:
(emilia.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3091  3091  2020-07-20T17:00:00  ...              29220        346535.0

[1 rows x 17 columns]

In [43]:
(veneto.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3107  3107  2020-07-20T17:00:00  ...              19649        457249.0

[1 rows x 17 columns]

In [44]:
(toscana.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3104  3104  2020-07-20T17:00:00  ...              10375        261958.0

[1 rows x 17 columns]

In [45]:
(liguria.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3094  3094  2020-07-20T17:00:00  ...              10105         91898.0

[1 rows x 17 columns]

In [46]:
(basilicata.sort_values('Deaths')).tail(1)

SNo                 Date  ... TotalPositiveCases  TestsPerformed
3088  3088  2020-07-20T17:00:00  ...                408         43486.0

[1 rows x 17 columns]

The value in "Deaths" is cumulative, so the row corresponding to the _most recent_ date contains the total number of deaths in the region.

### Death Rate
_total dead / total positive_

* Lombardia: **17.58%**
* Piemonte: **13.06%**
* Emilia-Romagna: **14.63%**
* Veneto: **10.43%**
* Toscana: **10.89%**
* Liguria: **10.49%**
* Basilicata: **6.96%**